### ISSUE TITLE
Propmt to save before optimisation/compilation (VS Code)

### ISSUE THREAD

**Issue by Woccz** [dbaumgarten/yodk#45]
Is there a way for VS Code NOLOL/YOLOL to prompt you to save the file before running the compiling/optimising scripts?
This isn't an urgent change, but it would be nice for quality of life if it is not difficult to implement.

Thank you.

**Comment by dbaumgarten** [comment]
That's a nice idea. I have stumbeled over this "action on unsaved file" thing relatively often myself (damn you, git merge-conflict resolving!!!).

The vscode-API probably provides a ways to do this. The only question is: how.

**Comment by dbaumgarten** [comment]
For now I have implemented it as:
```
if (vscode.window.activeTextEditor.document.isDirty){
await vscode.window.activeTextEditor.document.save()
}
```
In almost any case that should be what the user wants. I think displaying a prompt would just annoy him.
If he did save by accident he can just ctrl-Z...

Do you think otherwise?

**Comment by Woccz** [comment]
I thought to do a *save/no/do not ask me again* style prompt might be better if it is easy to implement?
That way the programmer can choose whether they want to save every time without being prompted.

Thank you for the current implementation.

**Comment by dbaumgarten** [comment]
Adding a prompt is probably be somewhat difficult ( I am not the most experienced vscode-typescript developer... ). Especially the "don't ask again", as it would need to save some state.

I think in 99.8% of the cases the user just forgot to save, because without saving the whole compile-command would be pretty useless. No need to annoy the user with a popup.

**Comment by Woccz** [comment]
Okay then that's fine too.

Thank you.

**Comment by dbaumgarten** [comment]
Published as part of v0.1.4
